### DISCUSSION TITLE
Using 'parent' as table field should also be reserved?

### DISCUSSION THREAD

**Discussion by Jackzmc** [PlutoLang/Pluto#244] [Q&A] [answered]
Apparently pluto now reserves the common word `parent`, which I don't think is ideal, but checking my code it fails with this vague error when I use `not data.parent`, removing that and it works fine:

```
$ plutoc -p jackz_vehicle_builder.lua
D:\Software\pluto\plutoc.exe: syntax error: jackz_vehicle_builder.lua:1462: expected an identifier.
    1462 | if not data.parent and subhandle ~= builder.base.handle then
         | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ here: this needs a name.
```

Besides the error reason being misleading/vague, is parent part of a field supposed to be reserved as well? I would think `local parent` would be reserved but not just any `<table>.parent`

I think that is way too broad of a keyword reservation

**Comment by Sainan** [comment, marked as answer]
`parent` has to be reserved for table constructors as otherwise the following would not be valid:
```Lua
class Entity
	__name = "Entity"
end

class Human extends Entity
	__name = "Human"

	function getHierarchy()
		return { self, parent }
	end
end

local h = new Human()

for h:getHierarchy() as t do
	print(t.__name)
end
```
As for shorthand accessor syntax, Pluto can remove this restriction: https://github.com/PlutoLang/Pluto/pull/245
